### DISCUSSION TITLE
Feature Request: Add option to keep text labels for taskbar items while always combining items on taskbar

### DISCUSSION THREAD

**Discussion by yetisyny** [valinet/ExplorerPatcher#202] [Ideas]
In Windows 10, you can have the taskbar set to “Always combine” items on the taskbar, and have it still display the text labels on the taskbar for running programs next to their icons.

Windows 11 by default always combines all items on the taskbar and also by default removes text labels for running programs next to their icons, only showing the icons.

In ExplorerPatcher’s current version, you have to have it “Never combine” or “Combine when taskbar is full” to restore the text labels next to the icons for running programs in Windows 11.

This is a problem for someone like me who wants it to always combine the items (to reduce clutter) but also wants the text labels brought back (so I have some idea what’s going on in the running programs besides just the icons), and who had that option in Windows 10 and wants to duplicate that functionality of Windows 10.

I propose adding a new item in ExplorerPatcher’s Properties window enabling or disabling text labels for running programs on the taskbar (text labels on = Windows 10 behavior, off = Windows 11 behavior). This would, for users who are upgrading from a previous version of ExplorerPatcher, have text labels turned off if things are set to always combine, and turned on otherwise. This would have to be shown twice of course, once for the main taskbar and once for other taskbars.

Another way to do it would be adding a 4th option to the “Combine items on main taskbar”/“Combine items on other taskbars” menu in ExplorerPatcher’s Properties window, and instead of just having “Always combine”, have “Always combine, icons only” (corresponding to Windows 11 behavior if ExplorerPatcher is not installed) and “Always combine, icons + text” (corresponding to Windows 10 behavior if you choose “Always combine” in Windows 10’s taskbar options).

I leave it up to the author of the program and other people in this discussion (if anyone else has a strong opinion) to decide which of these ways of doing it would be better, personally I am fine with either way.

Functionally, having a separate option regarding showing text labels for taskbar items would mean you’d now have 6 different possible combinations with that and the combining items on the taskbar option, instead of 4, and I doubt that many people would want to never combine or only combine when the taskbar is full, while having text labels disabled, because this would be a little weird, having multiple identical icons next to each other in the taskbar corresponding to different windows of the same program, without any text next to them explaining which window is which.

In earlier versions of Windows than Windows 10, if you opened up a very very large number of windows, eventually the taskbar would just be full of icons without text labels for each window, like I remember that sometimes happening in Windows XP as an example, however I found that pretty unpleasant. Then again, maybe some other people want that option. Although, to be clear, the classic Windows behavior from before Windows 10 is to never combine taskbar items and for the text labels to only gradually disappear the more windows you have open, so it takes a whole lot of open windows to get rid of the text labels completely. So having the text labels hidden right off the bat like in Windows 11, but not combining items on the taskbar, would be new behavior that doesn’t correspond to any option available in Windows 10, Windows 11, or any classic pre-10 version of Windows, either. But still it’s possible a small percentage of people might actually want the option to do things in this new way, have text labels hidden while not combining taskbar items (either never or only when the taskbar is full).

Anyway having 4 options for when to combine items on the taskbar, where you can have it always combine with text labels on or always combine with text labels off, as well as the options of never combining (text labels on) and only combining when the taskbar is full (text labels on), would duplicate all the functionality from past versions of Windows as well as Windows 11, without offering new functionality of having separate icons for each window without any text next to them, something I doubt many people would want but maybe a few people might want. You could actually have 6 options there, and also include options to never combine with text labels off and only combine when the taskbar is full while having text labels off.

The main thing is, I just want the ability to keep text labels for taskbar items while still having it set to always combine different windows of the same program into a single taskbar button, like I could do in Windows 10, so I can duplicate how I had things set up in Windows 10 and what I am used to. I liked the Windows 10 behavior when it was set to always combine taskbar items but where Windows 10 still kept the text labels, so mainly I’m just frustrated that in the current ExplorerPatcher version I can’t duplicate that behavior on Windows 11, since I’m trying to have Windows 11 behave the same way I’m used to Windows 10 behaving (as I had “Always combine” chosen in Windows 10 and it kept the text labels next to the icons).

I’ll let the author of the program decide the best way to restore that functionality from Windows 10, it doesn’t matter to me as long as I can have things that way (always combine taskbar items while keeping text labels on). I’m just putting some ideas out there about different ways this could be done in the ExplorerPatcher Preferences window. The simplest possible way to do it would be to have the option for showing or hiding text labels for taskbar items be a separate option, of course, but this would have to be listed twice, once for the main taskbar and once for other taskbars. If anyone else has any thoughts on this feature request, feel free to reply.

**Comment by valinet** [comment]
Hi

The behaviors EP exposes are the same ones from Windows 10 that have been there since the superbar was introduced in Windows 7, so in that regard, this program does its job, restoring what was previously available:
* never combine - each window presents a taskbar button with an icon and with an associated label besides it
* always combine - each program presents a taskbar button with an icon without any label besides it; same behavior as in Windows 11’s default taskbar
* combine when full - uses “never combine” until the taskbar gets too filled up and then switches to “always combine” but also showing a label besides (I think, never used that, not even in XP)

That’s everything the taskbar supported out of the box in Windows 10. There were 3 options and those 3 are all supported here. There was no “combine and show labels all the time” as far as I know. Logical or not those 3 options, that’s another thing to discuss, but that was what Microsoft was offering.

Anyway, I think what you describe could be achieved using 7+ taskbar tweaker, and that still is an option just fine in Windows 11 using EP: https://github.com/valinet/ExplorerPatcher/issues/9#issuecomment-939473986, but again, I don’t have much experience with that.

So, considering this, what is the actual feature request?

Thanks

**Comment by yetisyny** [comment]
Ah, I just checked on a Windows 10 computer and it turns out you’re correct about the 3 options in Windows 10 and you duplicate them accurately, I was misremembering things after upgrading to Windows 11, sorry. It seems I actually had it set up to combine taskbar buttons when full, not always, because I wanted to keep text labels, when I was using Windows 10.

Also I have installed the 7+ Taskbar Tweaker you suggested, but it doesn’t seem to offer this particular feature unfortunately (although its latest version does work on Windows 11 with your ExplorerPatcher).

Anyway what I was actually requesting was the ability to show text labels on the taskbar even when different windows for a program are combined (e.g. if you have multiple windows of File Explorer open, the text label says File Explorer when they get combined, instead of having the name of the folder like when they are separate, and the label is the name of the program, as currently happens when you are in the mode to combine buttons when the taskbar is full).

So basically, it would work the same as the way the option to combine taskbar buttons when full (the default from Windows 10) works, except it would always combine the buttons no matter what.

Since I was misremembering things from Windows 10, this is not important and it seems I messed up and requested a feature that I thought existed in Windows 10 but I was wrong (seems it doesn’t exist in Windows 10 and you accurately replicated Windows 10 behavior in Windows 11). Anyway, yeah I messed up by thinking this existed in Windows 10 when it didn’t. Sorry about that. I guess that would make it a new feature.

This isn’t really that important actually, it seems I misremembered things and your program is fine the way it is, I was just confused.

Anyway I guess this feature request is unnecessary unless a lot of other people want it too, what I’m requesting apparently never existed in Windows 10 and now that I checked things on a Windows 10 computer and confirmed your program accurately replicates its 3 options, this is not needed. Feel free to close this feature request as wontfix if you want, because I made it in error.

**Reply by valinet** [reply]
Yeah, no worries. If I discover a way to enable the behavior that you describe, be sure I won’t keep it to myself. I guess few bother since, if this was to be built-in, it would only show the name of the app anyway, so same info you kind of get from the icon as well. Now, to show the title of the last window used from that program, that’s another endeavor in its own.

Anyway, hopefully one day we get to implementing a proper custom replacement for the taskbar which we have total control of, so we could have all the options and flexibility in the world.

Thanks

**Comment by yetisyny** [comment]
I found out how to do this! This feature actually is in 7+ Taskbar Tweaker, it is one of the Advanced Options, the “show_labels” option can be set to 2 to always show labels, or set to 1 to always hide labels, or set to 0 for the default behavior of Windows. So this feature already exists in 7+ Taskbar Tweaker, I got it working. So if someone else wants to do this, they can just use both ExplorerPatcher and 7+ Taskbar Tweaker and use that in the Advanced Options.

I was confused at first but it is explained in the Help file for the Advanced Options, if you open 7+ Taskbar Tweaker, click the system icon in the upper left corner of the window for the system menu, click Advanced options to open the Advanced Options window, and click the Help button in the lower left corner of the Advanced Options window, then scroll down to the “show_labels” option in the list of options that are explained there. So if anyone else wants to do this, they can find it in that program. I guess this makes it unnecessary for you to do anything. Thanks for the recommendation of 7+ Taskbar Tweaker, which works very well alongside ExplorerPatcher.

**Reply by valinet** [reply]
I am glad that you sorted it out. Yeah, I knew that 7+TT has tons of options, the developer really took a lot of time to study the internal operation of the taskbar and how to alter it; I did not know specifically, as I don’t use it since ‘never combine’ is enough for me, but yeah, for even more advanced  use cases, it is the way to go.

And indeed, the tools can be used on their own no problem, which I think it is fine, no need to duplicate all the features in there in EP, I guess. 

Glad that you got it sorted out and thanks for sharing back your outcome.

**Comment by Gaurav-Original-ClassicShellTester** [comment]


**Reply by valinet** [reply]
Yeah, the world should be perfect.

Even if not as configurable as 7+TT, I think it would be a good start, especially if people are able to help by contributing as well.

Also, the research needed for 7+TT is not easy at all, it’s time consuming and beside moral reward, doesn’t bring much in return. Thus, as is the case with this project, probably the dev prioritizes features he/she uses, plus he/she has limited time to work on it anyway. So being surprised is something that gets beyond my understanding, if you consider these stuff more than the personal preference of features that you personally would like to see, plus the fact that we speak about a free project that someone develops during their free time, instead of riding a bike, swimming, reading books, working on more interesting stuff etc. just to fix some inconveniences a company you pay for their product refuses to fix. And that feature in particular is hard to get going on, as besides the usual explorer reverse engineer task, there’s also the hurdle of understanding the msstyles format, a totally undocumented Microsoft format which looks fine, when it works, to average people, but getting there from a developer perspective is very very hard work.

And as I keep saying, the solution to something not being implemented or implemented right is to get down to creating your own implementation. The resources are more than ever, there are places to learn off of.

**Reply by Gaurav-Original-ClassicShellTester** [reply]
I am sorry. What I meant was 7+TT is such a great app and its developer, like you with ExplorerPatcher, has also worked extremely hard to add every feature we can think of. :)
